backend/src/lib.rs

```diff
@@ -33,27 +38,20 @@ pub const DEFAULT_HALO2_MOCK_OPTIONS: &str = "";
 pub const DEFAULT_ESTARK_OPTIONS: &str = "stark_gl";
 
 impl BackendType {
-    pub fn factory<T: FieldElement>(&self) -> &'static dyn BackendFactory<T> {
-        #[cfg(feature = "halo2")]
-        const HALO2_FACTORY: halo2::Halo2ProverFactory = halo2::Halo2ProverFactory;
-        #[cfg(feature = "halo2")]
-        const HALO2_MOCK_FACTORY: halo2::Halo2MockFactory = halo2::Halo2MockFactory;
-        #[cfg(feature = "estark-polygon")]
-        const ESTARK_POLYGON_FACTORY: estark::polygon_wrapper::Factory =
-            estark::polygon_wrapper::Factory;
-        const ESTARK_STARKY_FACTORY: estark::starky_wrapper::Factory =
-            estark::starky_wrapper::Factory;
-        const ESTARK_DUMP_FACTORY: estark::DumpFactory = estark::DumpFactory;
-
+    pub fn factory<T: FieldElement>(&self) -> Box<dyn BackendFactory<T>> {
```

Comment:
I had to abandon the approach of creating `const` variables, because the `Halo2MockVadCop` one would have to be of type `VadCopWrapperFactory<T, Halo2MockFactory>`, but referencing `T` is not allowed!